packages/voyager-keycloak/src/KeycloakSecurityService.ts

```diff
@@ -90,9 +90,28 @@ export class KeycloakSecurityService implements SecurityService {
     }
   }
 
-  public async validateToken(token: string): Promise<boolean> {
-    const tokenObject = getTokenObject(token)
-    const result = await this.keycloak.grantManager.validateToken(tokenObject, 'Bearer')
-    return (result === tokenObject) ? true : false
+  public async onSubscriptionConnect(connectionParams: any, webSocket: any, context: any): Promise<any> {
+    const header = connectionParams.Authorization
+                  || connectionParams.authorization
+                  || connectionParams.Auth
+                  || connectionParams.auth
+    if (!header) {
+      throw new Error('Access Denied - missing Authorization field in connection parameters')
+    }
+    const token = this.getBearerTokenFromHeader(header)
+    try {
+      await this.keycloak.grantManager.validateToken(token, 'Bearer')
+      return token
+    } catch (e) {
+      this.log.error(`Error validating token from connectionParam ${header}\n${e}`)
+      throw new Error(`Access Denied - ${e}`)
+    }
+  }
+
+  private getBearerTokenFromHeader(header: any) {
```

Comment:
Original code pointed where we got this method. Let's put it here as well.

Reply:
This isn't taken from anywhere. I wrote it 😄 I will be putting some comments all over the place soon though.

Reply:
I have seen that before. Oh wait...
https://github.com/keycloak/keycloak-nodejs-connect/blob/214d576f3b013977eee664ff06ba955e358b043d/stores/bearer-store.js#L20-L31